dmsite/venv/src/store/actions/auth.js

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios';
+
+import * as actionTypes from './actionTypes';
+
+
+export const authStart = () => {
+  return {
+    type: actionTypes.AUTH_START,
+  };
+};
+
+export const authSuccess = (token) => {
+  return {
+    type: actionTypes.AUTH_SUCCESS,
+    token,
+  };
+};
+
+export const authFail = (error) => {
+  return {
+    type: actionTypes.AUTH_FAIL,
+    error,
+  };
+};
+
+export const logout = () => {
+  localStorage.removeItem('token');
+  localStorage.removeItem('expirationDate');
+  return {
+    type: actionTypes.AUTH_LOGOUT,
+  };
+};
+
+export const checkAuthTimeout = (expirationDate) => {
+  return (dispatch) => {
+    setTimeout(() => {
+      dispatch(logout);
+    }, expirationDate * 1000);
+  };
+};
+
+export const authLogin = (username, password) => {
+  return (dispatch) => {
+    dispatch(authStart());
+    axios.post('http://localhost:8000/rest-auth/login', {
+      username,
+      password,
+    })
+      .then((res) => {
+        const token = res.data.key;
+        const expDate = new Date(new Date().getTime() + 3600 * 1000);
```

Comment:
It may be a good idea to make 3600 a constant variable since you use it more than once.